core/phases/acting.py

```
@@ -0,0 +1,163 @@
+                if isinstance(result, str) and result.startswith("Error:"):
+                    raise ToolError(tool_name, result)
+                
+                if result:
+                    if attempt > 0:
+                        logger.info(f"Tool execution succeeded after {attempt + 1} attempts")
+                    return result
+                
+            except ToolError:
+                # 重新抛出 ToolError
+                raise
+            except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)